engine/src/main/java/io/camunda/zeebe/engine/processing/processinstance/CreateProcessInstanceProcessor.java

```diff
@@ -471,8 +473,13 @@ private void createEventSubscriptions(
       bpmnElementContext.init(
           elementInstanceKey, elementRecord, ProcessInstanceIntent.ELEMENT_ACTIVATED);
 
-      catchEventBehavior.subscribeToEvents(
-          bpmnElementContext, catchEventSupplier, sideEffectQueue, commandWriter);
+      final Either<Failure, ?> subscribedOrFailure =
+          catchEventBehavior.subscribeToEvents(
+              bpmnElementContext, catchEventSupplier, sideEffectQueue, commandWriter);
+
+      if (subscribedOrFailure.isLeft()) {
+        throw new UncheckedExecutionException(subscribedOrFailure.getLeft().getMessage());
```

Comment:
:wrench: We could enrich the error message with more context information.

If the error is logged then it is not obvious that the log statement is caused by a failed event subscription. The error message contains only the reason for the failure (e.g. `failed to evaluate expression 'unknown_var': no variable found for name 'unknown_var'`) but doesn't mention that the processor tries to create an event subscription for a scope. Even, if it's visible in the stacktrace.